<?xml version="1.0" ?>
<launch>
  <arg name="robot_x"   default="0.0" />
  <arg name="robot_y"   default="0.0" />
  <arg name="robot_yaw" default="0.0" />
  <arg name="prefix"    value="" />   <!-- this has to be "value", not "default", otherwise subst_value doesn't work -->
  <arg name="controller_name" default="my_controller" />
  <!-- Load URDF -->
  <include file="$(find mir_description)/launch/upload_mir_urdf.launch" />



  <remap from="base_footprint" to="$(arg controller_name)/base_footprint"/>
  <!-- Spawn the robot into Gazebo -->
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model $(arg controller_name)
    -x $(arg robot_x) -y $(arg robot_y) -Y $(arg robot_yaw) " />



  <!-- Load ros_control controller configurations -->
  <rosparam file="$(find mir_description)/config/joint_state_controller.yaml" command="load" />
  <rosparam file="$(find mir_description)/config/diffdrive_controller.yaml" command="load" subst_value="true" />

  <!-- Start the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
    args="joint_state_controller mobile_base_controller">
  </node>
 

  <!-- EKF -->
  <include file="$(find mir_gazebo)/launch/includes/ekf.launch.xml"/>

  <!-- Robot state publisher -->
   <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
  <param name="tf_prefix" value="$(arg controller_name)"/>
  </node>


<!-- create combined scan topic (like on real MiR) 
  <node pkg="topic_tools" type="relay" name="b_scan_relay" args="b_scan scan"/>
  <node pkg="topic_tools" type="relay" name="f_scan_relay" args="f_scan scan"/>

  <node name="b_rep117_laser_filter" pkg="mir_driver" type="rep117_filter.py" output="log">
    <remap from="scan" to="b_scan" />
    <param name="tf_prefix" value="$(arg controller_name)"/>
    <remap from="scan_filtered" to="b_scan_rep117" />
  </node>

  <node name="f_rep117_laser_filter" pkg="mir_driver" type="rep117_filter.py" output="log">
    <remap from="scan" to="f_scan" />
    <param name="tf_prefix" value="$(arg controller_name)"/>
    <remap from="scan_filtered" to="f_scan_rep117" />
  </node>

 -->

</launch>
